include/irr/asset/CTexturePacker.h

```diff
@@ -163,83 +133,144 @@ class ITexturePacker
-            const uint64_t bufOffset = static_cast<uint64_t>(m_megaimg->getCreationParameters().extent.width * m_megaimg->getCreationParameters().extent.height * (*offsets)[j].layer() + texOffset.y * m_megaimg->getCreationParameters().extent.width + texOffset.x)* texelSz;
+            const core::vector3du32_SIMD texOffset = pageCoords((*offsets)[j], m_pgSzxy);
+            const uint64_t bufOffset = static_cast<uint64_t>(m_physAddrTex->getCreationParameters().extent.width*m_physAddrTex->getCreationParameters().extent.height*texOffset.z + texOffset.y*m_physAddrTex->getCreationParameters().extent.width + texOffset.x) * texelSz;
 
             const uint8_t* src = reinterpret_cast<const uint8_t*>(_img->getBuffer()->getPointer())+reg.bufferOffset;
-            uint8_t* dst = reinterpret_cast<uint8_t*>(m_megaimg->getBuffer()->getPointer()) + bufOffset;
+            uint8_t* dst = reinterpret_cast<uint8_t*>(m_physAddrTex->getBuffer()->getPointer()) + bufOffset;
             const uint32_t pitch = reg.bufferRowLength*texelSz;
             for (uint32_t i = 0u; i < reg.imageExtent.height; ++i)
             {
-                memcpy(dst + ((reg.imageOffset.y + i)*m_megaimg->getCreationParameters().extent.width + reg.imageOffset.x)*texelSz, src+i*pitch, reg.imageExtent.width*texelSz);
+                memcpy(dst + ((reg.imageOffset.y + i)*m_physAddrTex->getCreationParameters().extent.width + reg.imageOffset.x)*texelSz, src+i*pitch, reg.imageExtent.width*texelSz);
             }
         }
+        //pack tail mips into 1 page (if needed)
+        if (offsets->back().hasMipTailAddr())
+        {
+            const core::vector3du32_SIMD texOffset = pageCoords((*offsets).back().getMipTailAddr(), m_pgSzxy);
+            //TODO
+        }
+        //fill page table
+        const VkExtent3D extent = _img->getCreationParameters().extent;
+        for (uint32_t i = 0u; i < offsets->size(); ++i)
+        {
+            const uint32_t w = std::max(extent.width>>(_subres.baseMipLevel+i),1u);
+            const uint32_t h = std::max(extent.height>>(_subres.baseMipLevel+i),1u);
+            const uint32_t pgSqrSz = core::roundUpToPoT((std::max(w,h)+m_pgSzxy-1u)/m_pgSzxy);
+
+            uint32_t* const pgTab = reinterpret_cast<uint32_t*>(
+                reinterpret_cast<uint8_t*>(m_pageTable->getBuffer()->getPointer()) + m_pageTable->getRegions().begin()[i].bufferOffset
+                );
+            const uint32_t pitch = m_pageTable->getRegions().begin()[i].bufferRowLength;
+            const uint32_t pgtH = m_pageTable->getRegions().begin()[i].imageExtent.height;
+            const uint32_t offset = (pitch*pgtH)*(*offsets)[i].layer() + (*offsets)[i].y()*pitch + (*offsets)[i].x();
+            for (uint32_t x = 0u; x < pgSqrSz; ++x)
+                for (uint32_t y = 0u; y < pgSqrSz; ++y)
+                    pgTab[offset + y*pitch + x] = (*offsets)[i].addr;
```

Comment:
yeah thats the wrong thing going into the page table,

I think i'll stop reviewing the commit here.

Reply:
you probably shouldnt even review this commit, look at state of code from last commit